### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ppth0608**
- Inference of return type
- Change the order of .success, .failure events of single trait

### CODE REVIEW

**Review by freak4pc — COMMENTED** [review]
I prefer leaving the types for type checking performance reasons here. I have no problem with the replacement of success and failure.

**Comment by ppth0608** [comment]
I think that omitting the Observable Element type will not affect the performance issue (ex. compile time), since the return type of the function is already defined.
I would appreciate any further comments on my comments. :)

File: Sources/Kingfisher+Rx.swift [hunk 1]
```diff
@@ -14,15 +14,15 @@ extension Reactive where Base == KingfisherWrapper<ImageView> {
     public func setImage(with resource: Resource?,
                          placeholder: Placeholder? = nil,
                          options: KingfisherOptionsInfo? = nil) -> Single<Image> {
-        return Single<Image>.create { [base] single in
+        return Single.create { [base] single in
```
Comment (kasyanov-ms) [inline Sources/Kingfisher+Rx.swift:17]: We could go further and completely remove the type declaration
```suggestion
        return .create { [base] single in
```

File: Sources/KingfisherManager+Rx.swift [hunk 2]
```diff
@@ -12,14 +12,14 @@ import Kingfisher
 extension Reactive where Base == KingfisherManager {
     public func retrieveImage(with resource: Resource,
                               options: KingfisherOptionsInfo? = nil) -> Single<Image> {
-        return Single<Image>.create { [base] single in
+        return Single.create { [base] single in
```
Comment (kasyanov-ms) [inline Sources/KingfisherManager+Rx.swift:15]: Similarly to https://github.com/RxSwiftCommunity/RxKingfisher/pull/11#discussion_r265926029
```suggestion
        return .create { [base] single in
```

File: Sources/Kingfisher+Rx.swift (hunk 1 above)
Comment (ppth0608) [reply Sources/Kingfisher+Rx.swift:17]: I agree to skip it completely if infer type 👍
We appreciate good feedback and reflect your feedback.

**Review by freak4pc — CHANGES_REQUESTED** [review]
Since this is not a bug, or there's nothing that this really "fixes", this boils down to a styling preference. As it is now, I'll be ok with `Single.create`, etc. but not the variation that drops the type completely.

Overall, I personally think that from a readability perspective dropping the entire type (e.g. `.create` only) is subpar. Without Xcode it makes code reviews much more annoying. 

FYI even if the return type is known, there is still some type inference performance cost to not specifying the generic type for a generic-heavy system such as Rx. A good example is when using flatMap - even if the containing method's return type is well-known, the flatMap expression, more times than not, requires an explicit return type. 

Thanks for your contribution!

**Comment by ppth0608** [comment]
@freak4pc I accepted the review content and rebased. I thought it would be nice to use abbreviations in Swift whenever possible, but when I saw the comments, I found that readability could be a problem.
`Single.create` may be a performance issue, but it does not seem to have a significant impact, and it seems to be good for readability.
I would like to contribute to RxKingfisher open source by adopting PR.

**Review by freak4pc — APPROVED** [review]
Looks good to me. Thanks for your contribution, welcome to RxSwift Community! :)

**Comment by ppth0608** [comment]
Thank you for accept my PR :)
If I get a chance, I will contribute again😄

**Pull request merged**